internal/transformer/transformresult_test.go

```diff
@@ -7,1585 +7,558 @@
-		t.Fatalf("Fail to transform read result, error: %v", err)
-	}
-	result, err := cv.Uint64Value()
-	if err != nil {
-		t.Fatalf("Fail to transform read result, error: %v", err)
-	}
-	if result != expected {
-		t.Fatalf("Unexpect test result, result '%v' should be '%v'", result, expected)
-	}
-	if cv.Type != v2.ValueTypeUint64 {
-		t.Fatalf("Unexpect test result, value type '%v' should be '%v'", cv.Type, v2.ValueTypeUint64)
-	}
-}
-
-func TestTransformReadResult_shift_should_be_valid(t *testing.T) {
-	val := uint64(1000000000)
-	shift := "-+4"
-	ro := models.ResourceOperation{DeviceResource: "test-object"}
-	cv, _ := dsModels.NewCommandValue(ro.DeviceResource, v2.ValueTypeUint64, val)
-	pv := models.PropertyValue{
-		Shift: shift,
-	}
-
-	err := TransformReadResult(cv, pv, lc)
-
-	if err == nil {
-		t.Fatalf("Unexpected test result, transform function should throw the error.")
+func Test_transformReadBase(t *testing.T) {
+	tests := []struct {
+		name        string
+		value       interface{}
+		base        string
+		expected    interface{}
+		expectedErr bool
+	}{
+		{"valid - uint8 base transformation", uint8(2), "10", uint8(100), false},
+		{"invalid - uint8 base transformation overflow", uint8(10), "10", nil, true},
+		{"valid - uint16 base transformation", uint16(2), "100", uint16(10000), false},
+		{"invalid - uint16 base transformation overflow", uint16(100), "10", nil, true},
+		{"valid - uint32 base transformation", uint32(2), "10000", uint32(100000000), false},
+		{"invalid - uint32 base transformation overflow", uint32(10000), "10", nil, true},
+		{"valid - uint64 base transformation", uint64(2), "100000", uint64(10000000000), false},
+		{"invalid - uint64 base transformation overflow", uint64(100000000), "10", nil, true},
+		{"valid - int8 base transformation", int8(2), "10", int8(100), false},
+		{"invalid - int8 base transformation overflow", int8(10), "10", nil, true},
+		{"valid - int16 base transformation", int16(2), "100", int16(10000), false},
+		{"invalid - int16 base transformation overflow", int16(100), "10", nil, true},
+		{"valid - int32 base transformation", int32(2), "10000", int32(100000000), false},
+		{"invalid - int32 base transformation overflow", int32(10000), "10", nil, true},
+		{"valid - int64 base transformation", int64(2), "100000", int64(10000000000), false},
+		{"invalid - int64 base transformation overflow", int64(100000000), "10", nil, true},
+		{"valid - float32 base transformation", float32(1.1), "2", float32(2.143547), false},
+		{"invalid - float32 base transformation overflow", math.MaxFloat32, "2", nil, true},
+		{"valid - float64 base transformation", float64(1.1), "2", float64(2.1435469250725863), false},
+		{"invalid - float64 base transformation overflow", math.MaxFloat64, "2", nil, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			res, err := transformReadBase(tt.value, tt.base)
+			if !tt.expectedErr {
+				assert.NoError(t, err)
+				switch tt.value.(type) {
```

Comment:
I don't see the need to this switch. assert.Equal() will handle passing both expected and actual as interface{}. If types don't match it will fail the equal test. So the whole switch can be replaced with 
```
assert.Equal(t, tt.expected, res)
```

Same in the other tests

Reply:
that's simple and nice, thanks for the tip!